internal/middleware/errorHandler.go

```diff
@@ -14,11 +19,17 @@ func ErrorHandler() gin.HandlerFunc {
 		if err == nil {
 			return
 		}
-		if c.Writer.Status() != http.StatusOK {
-			_, _ = c.Writer.WriteString(err.Error())
-			return
-		}
 
-		c.String(http.StatusInternalServerError, err.Error())
+		if errdef.IsDuplicated(err) {
+			_ = c.AbortWithError(http.StatusBadRequest, err)
+		} else if errdef.IsNotFound(err) {
+			_ = c.AbortWithError(http.StatusNotFound, err)
+		} else if errdef.IsUnauthorized(err) {
+			_ = c.AbortWithError(http.StatusUnauthorized, err)
+		} else {
+			id := uuid.New()
+			log.Printf("unhandled error: %v, log id: %s", err, id)
+			_ = c.AbortWithError(http.StatusInternalServerError, fmt.Errorf("unhandled error, log id: %s", id.String()))
```

Comment:
Is the error really unhandled? I mean we do have lots of error handling and the error middleware handling it 😄 I would at least not use that phrase towards users.

Maybe `Something went wrong. We'll look into it if you send us the id "21356" :)`

(If id implements the stringer interface then using the verb %s will lead to it being called. So calling `id.String()` yourself is not needed.)